Docs: Add YAML and Python example for context-aware recognizers

MRADULTRIPATHI opened the pull request:
## Change Description

This PR updates the analyzer docs to include examples for using context with custom YAML recognizers.  
The update clarifies how to define `context` in YAML and how to apply it in Python with `LemmaContextAwareEnhancer`.

This helps avoid confusion like in issue #1696 where context in YAML seemed ignored.

## Issue reference

Fixes #1696

## Checklist

- [x] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [x] I have signed the CLA
- [x] My code/docs update is limited to documentation only
- [x] All unit tests and lint checks pass locally
- [x] My PR contains documentation updates

## Example (YAML + Python)

See the added example in `developing_recognizers.md` for loading a DOB recognizer from YAML and applying context boosting in Python.

omri374 commented:
Hi @MRADULTRIPATHI, thanks for the PR. We have initial documentation for the YAML providers here: https://github.com/microsoft/presidio/blob/main/docs/analyzer/analyzer_engine_provider.md. How about we integrate the context examples there?

omri374 commented on docs/analyzer/analyzer_engine_provider.md:
```diff
@@ -174,3 +174,81 @@ nlp_configuration:
+    nlp_artifacts=nlp_artifacts,
+    recognizers=registry.recognizers,
+    context=["DOB"]
+)git checkout upstream/main -- docs/analyzer/adding_recognizers.md
```
```suggestion
)
```

omri374 commented on docs/analyzer/analyzer_engine_provider.md:
```diff
@@ -174,3 +174,81 @@ nlp_configuration:
+
+# Load recognizer from YAML
+registry = RecognizerRegistry()
+registry.add_recognizers_from_yaml("dob_recognizer.yml")
```
This assumes that the dob_recognizer is already saved to disk. Could you please add this step before calling this method?

MRADULTRIPATHI commented:
Hi @omri374,
Thanks for the review and helpful feedback!
I’ve updated the documentation accordingly:
Fixed the code block markers for both YAML and Python examples.
Removed the extra/unnecessary lines.
Added the explicit step to save dob_recognizer.yml before loading, so the example is self-contained and reproducible.
Please let me know if there’s anything else you’d like me to adjust.

omri374 commented:
Thanks! Could have been my bad, but the code chunks don't render correctly:

<img width="680" height="596" alt="image" src="https://github.com/user-attachments/assets/ce844654-c98e-4352-9390-e0bc366b33e6" />

MRADULTRIPATHI commented:
Hi @omrik⁩,
Thanks a lot for the quick and detailed review 🙏. I’ve updated the docs based on your feedback:
Fixed the YAML and Python code block formatting to render properly.
Verified that the context recognizer example and analyzer engine docs are now consistent.
Please let me know if anything else needs adjustment. Looking forward to your review and hopefully a merge

MRADULTRIPATHI commented:
Thanks @omri374 for the approval!  All checks have passed, so this PR should be ready to merge. 
Appreciate your time and guidance
